src/framework/handlers/untar.js

```diff
@@ -1,369 +1,202 @@
-
-// The UntarScope function is going to be used
-// as the code that ends up in a Web Worker.
-// The Untar variable is declared outside the scope so that
-// we do not have to add a 'return' statement to the UntarScope function.
-// We also have to make sure that we do not mangle 'Untar' variable otherwise
-// the Worker will not work.
-function UntarScope(isWorker) {
-    let utfDecoder;
-    let asciiDecoder;
-
-    if (typeof TextDecoder !== 'undefined') {
-        try {
-            utfDecoder = new TextDecoder('utf-8');
-            asciiDecoder = new TextDecoder('windows-1252');
-        } catch (e) {
-            console.warn('TextDecoder not supported - pc.Untar module will not work');
-        }
-    } else {
-        console.warn('TextDecoder not supported - pc.Untar module will not work');
-    }
-
-    function PaxHeader(fields) {
-        this._fields = fields;
-    }
-
-    PaxHeader.parse = function (buffer, start, length) {
-        const paxArray = new Uint8Array(buffer, start, length);
-        let bytesRead = 0;
-        const fields = [];
-
-        while (bytesRead < length) {
-            let spaceIndex;
-            for (spaceIndex = bytesRead; spaceIndex < length; spaceIndex++) {
-                if (paxArray[spaceIndex] === 0x20)
-                    break;
-            }
-
-            if (spaceIndex >= length) {
-                throw new Error('Invalid PAX header data format.');
-            }
-
-            const fieldLength = parseInt(utfDecoder.decode(new Uint8Array(buffer, start + bytesRead, spaceIndex - bytesRead)), 10);
-            const fieldText = utfDecoder.decode(new Uint8Array(buffer, start + spaceIndex + 1, fieldLength - (spaceIndex - bytesRead) - 2));
-            const field = fieldText.split('=');
-
-            if (field.length !== 2) {
-                throw new Error('Invalid PAX header data format.');
-            }
-
-            if (field[1].length === 0) {
-                field[1] = null;
-            }
-
-            fields.push({
-                name: field[0],
-                value: field[1]
-            });
-
-            bytesRead += fieldLength;
-        }
-
-        return new PaxHeader(fields);
-    };
-
-    PaxHeader.prototype.applyHeader = function (file) {
-        for (let i = 0; i < this._fields.length; i++) {
-            let fieldName = this._fields[i].name;
-            const fieldValue = this._fields[i].value;
-
-            if (fieldName === 'path') {
-                fieldName = 'name';
-            }
-
-            if (fieldValue === null) {
-                delete file[fieldName];
-            } else {
-                file[fieldName] = fieldValue;
-            }
-        }
-    };
+import { EventHandler } from '../../core/event-handler.js';
 
+/**
+ * An utility class for un-tar'ing archives, from a fetch requests.
+ * It processes files in tar in streamed manned, so assets parsing
+ * can happen in parallel instead of all at once at the end.
```

Comment:
```suggestion
 * A utility class for untaring archives, from a fetch request. It processes files from a tar files
 * in a streamed manner, so asset parsing can happen in parallel instead of all at once at the end.
```